src/mscorlib/src/System/Runtime/InteropServices/GcHandle.cs

```diff
@@ -197,10 +166,16 @@ public static GCHandle FromIntPtr(IntPtr value)
                     Mda.FireInvalidGCHandleCookieProbe(value);
                     return new GCHandle(IntPtr.Zero);
                 }
+                return new GCHandle(handle);
             }
 #endif
+            return new GCHandle(value);
+        }
 
-            return new GCHandle(handle);
+        internal static object GetTargetFromIntPtr(IntPtr value)
```

Comment:
The AssemblyLoadContext is not performance critical. The double verification is not a problem there.